pkg/cluster/purge_node_test.go

```diff
@@ -0,0 +1,164 @@
+package cluster
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientsetfake "k8s.io/client-go/kubernetes/fake"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+func Test_removeKubeadmEndpoint(t *testing.T) {
+	tests := []struct {
+		name                 string
+		endpoint             string
+		clusterResources     []runtime.Object
+		expectedRemovedIP    string
+		expectedRemainingIPs []string
+		expectedConfigMap    corev1.ConfigMap
+		wantErr              bool
+	}{
+		{
+			name:     "when endpoint is removed expect kubeadm configmap to be mutated",
+			endpoint: "master-test-2", // removing master-test-2
+			clusterResources: []runtime.Object{
+				&corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      kubeadmconstants.KubeadmConfigConfigMap,
+						Namespace: metav1.NamespaceSystem,
+					},
+					TypeMeta: metav1.TypeMeta{},
+					Data: map[string]string{
+						clusterStatusConfigMapKey: `kind: ClusterStatus
+apiVersion: kubeadm.k8s.io/v1beta2
+apiEndpoints:
+  master-test-1:
+    advertiseAddress: 10.128.0.126
+    bindPort: 6443
+  master-test-2:
+    advertiseAddress: 10.128.0.63
+    bindPort: 6443
+`,
```

Comment:
These raw string literals are pragmatic but could use an aesthetic improvement. Open to suggestions.